clippy_lints/src/deprecated_lints.rs

```diff
@@ -123,3 +123,13 @@ declare_deprecated_lint! {
     pub UNSAFE_VECTOR_INITIALIZATION,
     "the replacement suggested by this lint had substantially different behavior"
 }
+
+
+/// **What it does:** Nothing, this lint has been deprecated.
+///
+/// **Deprecation reason:** This lint was merged with new_without_default
+declare_deprecated_lint! {
```

Comment:
Could you register it as a renamed lint instead? Have a look at https://github.com/rust-lang/rust-clippy/pull/3554 to how that's done